### PR TITLE
add support for generating xml documentation

### LINKED ISSUE #312: DocumentationMode compileOptions
From https://github.com/aspnet/dnx/issues/3189

### PR DESCRIPTION
**Pull request by anurse**
fixes #312 (once this change is in, I'll also be fixing #635) 

/cc @victorhurdugaci @davidfowl

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectModel/ProjectReader.cs [hunk 1]
```diff
@@ -535,6 +535,7 @@ private static CommonCompilerOptions GetCompilationOptions(JsonObject rawObject)
                 DelaySign = rawOptions.ValueAsNullableBoolean("delaySign"),
                 PublicSign = rawOptions.ValueAsNullableBoolean("publicSign"),
                 EmitEntryPoint = rawOptions.ValueAsNullableBoolean("emitEntryPoint"),
+                GenerateXmlDocumentation = rawOptions.ValueAsNullableBoolean("generateXmlDocumentation"),
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/ProjectReader.cs]: can we pick a less bad name? One that matches the csc flag?

Comment (anurse) [inline src/Microsoft.DotNet.ProjectModel/ProjectReader.cs]: The CSC flag is `doc`, what do you suggest? "`doc`: true`"?

**Comment by anurse** [comment]
:up: :date:

**Comment by davidfowl** [comment]
includeDoc

**Comment by davidfowl** [comment]
This is file :shipit:

**Comment by anurse** [comment]
I prefer `xmlDoc` to `includeDoc` so I'm going to merge it with that. Just waiting of verification of the change and I'll merge and move on to #635

**Pull request closed without merging**